x-pack/plugins/rule_registry/server/rule_data_plugin_service/rule_data_plugin_service.ts

```diff
@@ -105,6 +106,8 @@ export class RuleDataPluginService {
       indexOptions,
     });
 
+    this.registeredIndices.set(indexOptions.registrationContext, indexInfo);
```

Comment:
The full index name (`indexInfo.baseName`) is defined not only by `registrationContext`, but also by `dataset` (`alerts` or `events`).

I guess the proper key for this map would be `${indexOptions.registrationContext}.${indexOptions.dataset}`, and `getRegisteredIndexInfo` would need to accept `dataset` as well.

Reply:
True, this only works because we assume `alerts` as a dataset implicitly. Would it be ok to add that as a follow-up to unblock this ASAP?

Reply:
Yep, of course 👍 
TODO in the follow-up PR: let `getRegisteredIndexInfo` to accept `dataset` as well, generate keys of the `registeredIndices` map correctly.